src/redux/slices/keyring.js

```diff
@@ -232,6 +213,7 @@ export const setCurrentPrincipal = createAsyncThunk(
       const state = getState();
       const { instance } = state.keyring;
       await instance?.setCurrentPrincipal(walletNumber);
+      await instance?.getICNSData();
```

Comment:
are you doing something with the data?

Reply:
Yes i need that ICNSData so when you change the current, you can see your icns name directly in the accouts modals. Also if you'r changing the principal, it's correct to fetch the icnsData for the new currentWallet